core/src/main/scala/kafka/coordinator/group/GroupMetadataManager.scala

```diff
@@ -565,6 +644,7 @@ class GroupMetadataManager(brokerId: Int,
 
         for (group <- groupMetadataCache.values) {
           if (partitionFor(group.groupId) == offsetsPartition) {
+            group.activeProducers.foreach(removeProducerGroup(_, group.groupId))
```

Comment:
I don't think accessing `activeProducers` is safe without the group lock. The call to `onGroupUnloaded` will transition the state of the group to `Dead`. Maybe after that, we can just iterate through `openGroupsForProducer` and remove the group from every producer.